torchdata/datapipes/iter/transform/bucketbatcher.py

```diff
@@ -92,7 +119,109 @@ def __new__(
         if sort_key is not None:
             # In-batch shuffle each bucket seems not that useful, it seems misleading since .batch is called prior.
             if in_batch_shuffle:
-                datapipe = datapipe.batch(batch_size=bucket_num, drop_last=False).map(fn=_in_batch_shuffle_fn).unbatch()
+                datapipe = datapipe.batch(batch_size=bucket_num, drop_last=False).in_batch_shuffle().unbatch()
             else:
                 datapipe = datapipe.shuffle(buffer_size=bucket_size)
         return datapipe
+
+
+def _default_len_fn(token):
+    return len(token)
+
+
+def _token_len_fn(token, len_fn):
+    return len_fn(token), token
+
+
+def _token_filter_fn(data, *, min_len, max_len):
+    length, token = data
+    return length >= min_len and length <= max_len
+
+
+@functional_datapipe("max_token_bucketize")
+class MaxTokenBucketizerIterDataPipe(IterDataPipe[DataChunk[T_co]]):
+    r"""
+    Creates mini-batches of data from a min-heap with limited size, and the total length of samples
+    returned by ``len_fn`` within each batch will be limited by ``max_token_count``.
+    (functional name: ``max_token_bucketize``). If ``min_len`` or ``max_len`` is set, the samples with
```

Comment:
Nit: Does the docgen look right? It is still after the `.`

As long as the doc looks right, this is fine

Reply:
Good catch.